pkg/dynamicaccess/accesslogic.go

```diff
@@ -11,54 +11,65 @@ import (
 
 var hashFunc = sha3.NewLegacyKeccak256
 
-// Logic has the responsibility to return a ref for a given grantee and create new encrypted reference for a grantee
-type Logic interface {
+// Read-only interface for the ACT
+type Decryptor interface {
+	// DecryptRef will return a decrypted reference, for given encrypted reference and grantee
+	DecryptRef(rootHash swarm.Address, encryped_ref swarm.Address, publisher *ecdsa.PublicKey) (swarm.Address, error)
+	// Embedding the Session interface
+	Session
```

Comment:
what is the purpose of this embedded IF here ? Check "Avoid Embedding Types in Public Structs" part of the coding style

Reply:
We were told by Viktor to do it this way. I think the reason is that we can access the functions of the Session this way.